ui/hooks/metamask-notifications/useSwitchNotifications.test.tsx

```diff
@@ -1,125 +1,183 @@
-import React from 'react';
-import { Provider } from 'react-redux';
-import { renderHook, act } from '@testing-library/react-hooks';
-import configureStore from 'redux-mock-store';
-import thunk from 'redux-thunk';
-import type { Store } from 'redux';
-import * as actions from '../../store/actions';
-import { MetamaskNotificationsProvider } from '../../contexts/metamask-notifications/metamask-notifications';
+import { waitFor } from '@testing-library/react';
+import * as ActionsModule from '../../store/actions';
+import * as NotificationSelectorsModule from '../../selectors/metamask-notifications/metamask-notifications';
+import { renderHookWithProviderTyped } from '../../../test/lib/render-helpers';
 import {
   useSwitchFeatureAnnouncementsChange,
-  useSwitchAccountNotifications,
   useSwitchAccountNotificationsChange,
+  useAccountSettingsProps,
 } from './useSwitchNotifications';
 
-const middlewares = [thunk];
-const mockStore = configureStore(middlewares);
-
-jest.mock('../../store/actions', () => ({
-  setFeatureAnnouncementsEnabled: jest.fn(),
-  checkAccountsPresence: jest.fn(),
-  updateOnChainTriggersByAccount: jest.fn(),
-  deleteOnChainTriggersByAccount: jest.fn(),
-  showLoadingIndication: jest.fn(),
-  hideLoadingIndication: jest.fn(),
-  fetchAndUpdateMetamaskNotifications: jest.fn(),
-}));
-
-describe('useSwitchNotifications', () => {
-  let store: Store;
-
+describe('useSwitchFeatureAnnouncementsChange() tests', () => {
   beforeEach(() => {
-    store = mockStore({
-      metamask: {
-        isFeatureAnnouncementsEnabled: false,
-        internalAccounts: {
-          accounts: {
-            '0x123': {
-              address: '0x123',
-              id: 'account1',
-              metadata: {},
-              options: {},
-              methods: [],
-              type: 'eip155:eoa',
-            },
-          },
-        },
-      },
-    });
-
-    store.dispatch = jest.fn().mockImplementation((action) => {
-      if (typeof action === 'function') {
-        return action(store.dispatch, store.getState);
-      }
-      return Promise.resolve();
-    });
+    jest.restoreAllMocks();
+  });
 
-    jest.clearAllMocks();
+  const arrangeMocks = () => {
+    const mockSetFeatureAnnouncementsEnabled = jest.spyOn(
+      ActionsModule,
+      'setFeatureAnnouncementsEnabled',
+    );
+    return {
+      mockSetFeatureAnnouncementsEnabled,
+    };
+  };
+
+  it('should update feature announcement when callback invoked', async () => {
+    const mocks = arrangeMocks();
+    const hook = renderHookWithProviderTyped(
+      () => useSwitchFeatureAnnouncementsChange(),
+      {},
+    );
```

Comment:
replaced mocking store and dispatch to instead use `renderHookWithProviderTyped` which does the setup for you.

The `{}` in the second param is to let the function add redux provider (we could instead pass in a partial store, but it is unneeded)